phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -576,20 +597,27 @@ public static void wrapResultUsingOffset(final RegionCoprocessorEnvironment envi
                 }
             }
             Result joinResult = null;
-            if (dataRegion != null) {
-                joinResult = dataRegion.get(get);
-            } else {
-                TableName dataTable =
-                    TableName.valueOf(MetaDataUtil.getLocalIndexUserTableName(environment.getRegion().
-                        getTableDescriptor().getTableName().getNameAsString()));
-                Table table = null;
-                try {
-                    table = environment.getConnection().getTable(dataTable);
+            if (ScanUtil.isLocalIndex(scan)) {
+                if (dataRegion != null) {
+                    joinResult = dataRegion.get(get);
+                } else {
+                    TableName dataTable =
+                            TableName.valueOf(MetaDataUtil.getLocalIndexUserTableName(
+                                    environment.getRegion().getTableDescriptor().getTableName().
+                                            getNameAsString()));
+                    try (Table table = environment.getConnection().getTable(dataTable)) {
+                        joinResult = table.get(get);
+                    }
+                }
+            } else if (ScanUtil.isUncoveredGlobalIndex(scan)) {
+                byte[] dataTableName = scan.getAttribute(PHYSICAL_DATA_TABLE_NAME);
```

Comment:
Can we have a test where the index table has a different Physical Table name and data table has a different physical table name?

Reply:
Not sure if I understood this comment. Data and index tables have different physical tables always.

Reply:
This code reminded me a test case where we have separate physical and logical table names for (case1. data table and case2. index table). I was just asking to make sure that we are not breaking anything.
Something like this: 
You have an index hint for a global index which has a different physical table name (logical name appears in the hint but its PHYSICAL_TABLE_NAME in syscat points to different hbase table). I think your code doesn't break this case but I wanted to make sure. Does this make sense?

Reply:
I see. I think this concern is outside the context of this PR as this PR does not change the existing index hint implementation or how we pass the data table physical name via a scan attribute.